### PR TITLE
Add two `otelarrowcol` examples; add `experimentprocessor`, move generated components

### PR DESCRIPTION
**Pull request by jmacd**
The current `collector` subdirectory contains only generated components. I wished to begin adding new components, so moved the generated code from its current location into `./collector/gen`. 

The `otelarrowcol` command was copied from `otelcorecol`. This is now derived code, instead of being copied. This collector main() adds the new experiment collector included in this PR. (Why: the Makefile for OTC manages this process, particularly versioning, and I do not want to duplicate it in this repo at this time in order to continue using the OTC "builder".)

A new OTC processor named `experimentprocessor` is written with basic test coverage. TODO: improve test coverage.

Two new examples are placed in the ./collector/examples directory. 

1. A bridge example (two configs)
2. A loopback example (one config) using the experimentprocessor.

### CODE REVIEW

File: collector/processor/experimentprocessor/config.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	// validate that there's at least one item in the table
+	if len(c.Table) == 0 {
+		return fmt.Errorf("invalid route table: %w", errNoTableItems)
+	}
+
+	// validate each route table item
+	total := 0
+	for _, item := range c.Table {
+		if item.Weight < 0 {
+			return fmt.Errorf("invalid route weight: %d: %w", item.Weight, errInvalidWeight)
```
Comment (lquerel) [inline collector/processor/experimentprocessor/config.go:53]:
```suggestion
			return fmt.Errorf("invalid route weight %d: %w", item.Weight, errInvalidWeight)
```

**Review by lquerel — APPROVED** [review]
LGTM

File: collector/examples/bridge/edge-collector.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+exporters:
+  # otlp/out configures an arrow exporter
+  otlp/out:
+    endpoint: 0.0.0.0:5000
+    tls:
+      insecure: true
+    arrow:
+      enabled: true
+      num_streams: 1
```
[the remark this answers is not in the document]
Comment (jmacd) [reply collector/examples/bridge/edge-collector.yaml]: I think we could change the default to use NumCPUs(). For now I've updated the comment to recommend a limit of the number of CPUs.

[the remark this answers is not in the document]
Comment (jmacd) [reply collector/examples/bridge/edge-collector.yaml]: https://github.com/open-telemetry/experimental-arrow-collector/issues/31

**Pull request merged**